reactioncommerce/reaction
# (fix): set slugified title as permalink

Resolves #4735 
Impact: **major**  
Type: **bugfix**

## Issue
When you create a product, the permalink (handle) is set to a random ID instead of the slugified version of the title. This is happening because:
- the code initially sets the handle to a random ID (in the meanwhile, the title is null)
- when the title is set the handle is not updated because the condition `doc.handle === doc._id` is never true (the handle is set to a random ID, not the doc ID)

Steps to reproduce:
1. Create a new product
2. Observe that the handle is not a slugified version of the title
3. Update the title -> the handle is not updated

## Solution
I removed the `doc.handle === doc._id` condition. This way when the user sets a Title for the product, the handle is updated. I also removed the initial population with a random ID because I see no use for it. 

## Breaking changes
None

## Testing
1. Log in the Operator UI and create a product
2. Both the Title and the Handle are null
3. Set a Title for the product and tab to another field
4. Notice how the Handle is updated to the title's slugified version
5. Keep changing the Title and notice how the Handle keeps getting updated each time
